Streetcode/Streetcode.XUnitTest/MediatRTests/StreetCode/RelatedTerm/GetAllByTermId/GetAllRelatedTermsByTermIdHandlerTest.cs

```diff
@@ -0,0 +1,121 @@
+﻿using AutoMapper;
+using FluentResults;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore.Query;
+using Moq;
+using Streetcode.BLL.DTO.Streetcode.TextContent;
+using Streetcode.BLL.MediatR.Streetcode.RelatedTerm.GetAllByTermId;
+using Streetcode.DAL.Repositories.Interfaces.Base;
+using System.Linq.Expressions;
+using Xunit;
+using Entity = Streetcode.DAL.Entities.Streetcode.TextContent.RelatedTerm;
+
+namespace Streetcode.XUnitTest.MediatRTests.StreetCode.RelatedTerm.GetAllByTermId
+{
+    public class GetAllRelatedTermsByTermIdHandlerTest
+    {
+        private Mock<IRepositoryWrapper> _mockRepository;
+        private Mock<IMapper> _mockMapper;
+
+        public GetAllRelatedTermsByTermIdHandlerTest()
+        {
+            _mockRepository = new Mock<IRepositoryWrapper>();
+            _mockMapper = new Mock<IMapper>();
+        }
+
+
+        [Theory]
+        [InlineData(1)]
+        public async Task Handle_ReturnsError_WhenRepositoryReturnsNull(int id)
+        {
+            // Arrange
+            var query = new GetAllRelatedTermsByTermIdQuery(id);
+            _mockRepository.Setup(x => x.RelatedTermRepository
+            .GetAllAsync(It.IsAny<Expression<Func<Entity, bool>>>(),
+                It.IsAny<Func<IQueryable<Entity>, IIncludableQueryable<Entity, object>>>()))
+                .ReturnsAsync((IEnumerable<Entity>)null);
+            var handler = new GetAllRelatedTermsByTermIdHandler(_mockMapper.Object, _mockRepository.Object);
+
+            // Act
+            var result = await handler.Handle(query, CancellationToken.None);
+
+            // Assert
+            Assert.Equal("Cannot get words by term id", result.Errors.First().Message);
+        }
+
+        [Theory]
+        [InlineData(1)]
+        public async Task Handle_ReturnsError_WhenMapperReturnsNull(int id)
+        {
+            // Arrange
+            var query = new GetAllRelatedTermsByTermIdQuery(id);
+            var relatedTerms = new List<Entity> { new Entity() };
+            _mockRepository.Setup(x => x.RelatedTermRepository.
+            GetAllAsync(It.IsAny<Expression<Func<Entity, bool>>>(),
+                It.IsAny<Func<IQueryable<Entity>, IIncludableQueryable<Entity, object>>>()))
+                .ReturnsAsync(relatedTerms);
+            _mockMapper.Setup(x => x.Map<IEnumerable<RelatedTermDTO>>(relatedTerms))
+                .Returns((IEnumerable<RelatedTermDTO>)null);
+            var handler = new GetAllRelatedTermsByTermIdHandler(_mockMapper.Object, _mockRepository.Object);
```

Comment:
i would recommend some extra spaces between operatrions to make code more readable

Reply:
Done